convert2rhel/checks.py

```diff
@@ -447,3 +456,57 @@ def _bad_kernel_substring(kernel_release):
         )
         return True
     return False
+
+
+def check_package_updates():
+    """Ensure that the system packages installed are up-to-date."""
+    logger.task("Prepare: Checking if the packages are up-to-date")
+
+    # Get a total count of how many packages is left to update
+    total_packages = get_total_packages_to_update()
+
+    if total_packages > 0:
+        logger.warning(
+            "The system has %s packages to be updated. "
+            "To proceed with the conversion, update the packages on your system by executing the following step:\n\n"
+            "Run: yum update -y\n" % total_packages
+        )
+    else:
+        logger.info("System is up-to-date.")
+
+
+def is_loaded_kernel_latest():
+    """Check if the loaded kernel is behind or of the same version as in yum repos."""
+    logger.task("Prepare: Checking if the loaded kernel version is the most recent.")
+
+    # The latest kernel version on repo
+    repoquery_output, _ = run_subprocess('repoquery -q --qf "%{BUILDTIME}\\t%{VERSION}-%{RELEASE}" kernel')
```

Comment:
@r0x0d @oamg/convert2rhel-developers   what in case the repository is not reachable? e.g.
* detached systems
* disabled repositories

in such a case you will not obtain any information to be able to compare the versions. (we already hit such a systems multiple times). unless you have written in requirements to have enabled base repositories for the particular system (for OL, CentOS, ..). If not, I suggest to have a fallback to check the latest installed kernel as well.

Additionally, for the installed packages, for EL 8+ you need to care about `kernel-core` rpms instead of `kernel`, as the `kernel` rpm is just a metapackage since EL 8 and it doesn't have to be installed at all (this is not problem when you check what's newest pkg in repository - if repo is enabled and accessible)

Reply:
to simulate such a situation, here is my output from centos 7 container
```
[root@1c732f612637 /]# repoquery -q --qf "%{BUILDTIME}\\t%{VERSION}-%{RELEASE}" --disablerepo="*" kernel
[root@1c732f612637 /]# echo $?
0
```

Reply:
Hey, @pirat89! Thanks for your insights on this. Just out of curiosity, how did you guys handle those cases you mentioned? I don't recall having a programmatic set of enabling repositories in the tool, as far as I know, having at least the base repositories enabled is a user requirement (But I may be wrong on that). 

@SpyTec do you know any cases where the base repos for a system are disabled for our users?

Reply:
@pirat89 From our stand-up and GChat, we require repos to be enabled as mentioned in the docs. So not an issue we need to worry about for this PR

> Access to repositories providing packages from the original distribution is required, especially to the base repository, to perform a rollback. 

https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html-single/converting_from_an_rpm-based_linux_distribution_to_rhel/index#con_rollback_converting-from-a-linux-distribution-to-rhel

Reply:
Hi guys, sorry for late response. I am not looking here so much during my holiday. I've read the discussion and it seems you should not have any problems here regarding the hard requirement to have enabled the base repository on the system. The unsupported envar seems as a good decision to bypass that requirement, keeping the possibility to handle problems by customers on their own.

@r0x0d  I spoke about cases we discovered in case of in-place upgrades between major versions of RHEL systems - probably I should mention that, sorry for the confusion. I am trying to inform you about problems we already had to deal with when I see you are dealing with similar problems :)